p2p/peer_behaviour.go

```diff
@@ -0,0 +1,89 @@
+package p2p
+
+type ErrorBehaviourPeer int
+
+const (
+	ErrorBehaviourUnknown = iota
+	ErrorBehaviourBadMessage
+	ErrorBehaviourMessageOutofOrder
+)
+
+type GoodBehaviourPeer int
+
+const (
+	GoodBehaviourVote = iota
+	GoodBehaviourBlockPart
+)
+
+// PeerBehaviour provides an interface for reactors to signal the behaviour
+// of peers synchronously to other components.
+type PeerBehaviour interface {
+	Behaved(peer Peer, reason GoodBehaviourPeer)
+	Errored(peer Peer, reason ErrorBehaviourPeer)
+}
+
+type switchedPeerBehaviour struct {
+	sw *Switch
+}
+
+func (spb *switchedPeerBehaviour) Errored(peer Peer, reason ErrorBehaviourPeer) {
+	spb.sw.StopPeerForError(peer, reason)
+}
+
+func (spb *switchedPeerBehaviour) Behaved(peer Peer, reason GoodBehaviourPeer) {
+	spb.sw.MarkPeerAsGood(peer)
+}
+
+func NewswitchedPeerBehaviour(sw *Switch) PeerBehaviour {
+	return &switchedPeerBehaviour{
+		sw: sw,
+	}
+}
+
+type ErrorBehaviours map[Peer][]ErrorBehaviourPeer
+type GoodBehaviours map[Peer][]GoodBehaviourPeer
```

Comment:
As those are public they need proper doc strings.

I'd also advice to store `ID` instead of `Peer` to continue with our ambitions to not depend on an entire `Peer` for the behavioural part.